src/tree/registerTrees.ts

```diff
@@ -42,13 +44,17 @@ export function registerTrees(): void {
     /* eslint-disable-next-line @typescript-eslint/promise-function-async */
     registerCommand(imagesLoadMore, (context: IActionContext, node: AzExtTreeItem) => ext.imagesTree.loadMore(node, context));
 
-    ext.registriesRoot = new RegistriesTreeItem();
-    const registriesLoadMore = 'vscode-docker.registries.loadMore';
-    ext.registriesTree = new AzExtTreeDataProvider(ext.registriesRoot, registriesLoadMore);
-    ext.registriesTreeView = vscode.window.createTreeView('dockerRegistries', { treeDataProvider: ext.registriesTree, showCollapseAll: true, canSelectMany: false });
-    ext.context.subscriptions.push(ext.registriesTreeView);
+    const urtdp = new UnifiedRegistryTreeDataProvider(ext.context.globalState);
+    const genericRegistryV2DataProvider = new GenericRegistryV2DataProvider(ext.context);
+    urtdp.registerProvider(new GitHubRegistryDataProvider(ext.context));
+    urtdp.registerProvider(new DockerHubRegistryDataProvider(ext.context));
+    urtdp.registerProvider(new AzureRegistryDataProvider(ext.context));
+    urtdp.registerProvider(genericRegistryV2DataProvider);
+    ext.registriesRoot = urtdp;
+    ext.registriesTreeView = vscode.window.createTreeView('dockerRegistries', { treeDataProvider: urtdp });
+    ext.registriesTree = urtdp;
+    // TODO: add gitlab registry provider when the new extension is ready
```

Comment:
GitLab's provider will be added by the other extension, no code will be needed in this extension for that

Reply:
ohh that makes sense